### PR TITLE
[containerapp] Update api-version to 2023-08-01-preview

### PR DESCRIPTION
**Pull request by Greedygre**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
containerapp

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_workload_profile_commands.py [hunk 1]
```diff
@@ -376,14 +376,13 @@ def test_containerapp_env_enable_workload_profiles_infer_env_type(self, resource
             JMESPathCheck("properties.workloadProfiles", None),
         ])
 
-    @live_only()
     @AllowLargeResponse(8192)
     @ResourceGroupPreparer(location="eastus")
     def test_containerapp_create_enable_dedicated_gpu(self, resource_group):
-        self.cmd('configure --defaults location={}'.format("westus3"))
-        env = self.create_random_name(prefix='env', length=24)
+        self.cmd('configure --defaults location={}'.format("northeurope"))
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_workload_profile_commands.py:382]: remove the config and config the resource group location to target region

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_patch.py [hunk 2]
```diff
@@ -151,14 +151,13 @@ def test_containerapp_patch_list_and_apply_without_arguments_e2e(self, resource_
             create_and_verify_containerapp_up(self, resource_group, source_path=source_path, ingress=ingress, target_port=target_port, app_name=app_name)
 
             # Execute and verify patch list command
-            self.cmd(f'configure --defaults group={resource_group}')
-            patchable_images = self.cmd(f'containerapp patch list').get_output_in_json()
+            patchable_images = self.cmd(f'containerapp patch list -g {resource_group}').get_output_in_json()
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_patch.py:154]: keep config

**Review by Juliehzl — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 

Could you please help to review and merge this PR? As there are other PR wait for this pr for updating api-version to 2023-08-01-preview, thanks for your help!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**